lib/pharos/salt_minion.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+require "pharos/salt_api"
+
+module Pharos
+  # This class represents a salt minion
+  class SaltMinion
+    include SaltApi
+
+    # Initializes a new salt minion identified by mid.
+    def initialize(mid:)
+      @mid = mid
+    end
+
+    # Return information for this minion.
+    def info
+      res = perform_request(endpoint: "/minions/#{@mid}", method: "get")
+      JSON.parse(res.body)["return"].first[@mid]
+    end
+
+    # Check if this minion has any role assigned.
+    def roles?
+      !info["roles"].blank?
+    end
+
+    # Map our high-level roles to more salt specialized roles.
+    def map_role(role:)
```

Comment:
[MINOR] Maybe we can replace this method with a constant:

```
ROLES_MAP = {
  master: ["kube-master", "etcd"],
  minion: ["kube-minion"]
}

Calling ```ROLES_MAP[:master]``` or ```map_role(role: :master)``` isn't so different but we avoid a method invocation and tests for trivial things.
```